R/cut.R

```diff
@@ -0,0 +1,129 @@
+#' Cut a numeric Crunch variable
+#'
+#' `crunch::cut()` is equivalent to `base::cut()` except that it operates on
+#' Crunch variables instead of in-memory variables. The function divides the range of
+#' `x` into intervals and codes the values in x according to which interval they fall.
+#' The leftmost interval corresponds to level one, the next leftmost to level two and
+#' so on.
+#' @param x A crunch variable of the class NumericVariable
+#' @param breaks Either a numeric vector of two or more unique cut points
+#' or a single number (greater than or equal to 2) giving the number of intervals
+#' into which x is to be cut.
+#' @param name The name of the resulting case variable as a character string.
+#' @param labels labels for the levels of the resulting category.
+#' By default, labels are constructed using interval notation.
+#' If labels = FALSE, simple integer codes are returned instead of a factor.
+#' @param include.lowest logical, indicating if an `x[i]` equal to the lowest
+#' (or highest, for right = FALSE) `breaks` value should be included.
+#' @param right logical, indicating if the intervals should be closed on the right
+#' (and open on the left) or vice versa.
+#' @param dig.lab	integer which is used when labels are not given.
+#' It determines the number of digits used in formatting the break numbers.
+#' @param ordered_result	Ignored.
+#' @param ... further arguments passed to [makeCaseVariable]
+#' @return a Crunch [`VariableDefinition`]. Assign it into the dataset to create
+#' it as a derived variable on the server.
+#' @export
+#' @examples
+#' \dontrun{
+#' ds <- loadDataset("mtcars")
+#' ds$cat_var <- cut(ds$mpg, 3, variableName = "new_var")
+#' }
+#'
+setMethod("cut", "NumericVariable", function(x,
+                                             breaks,
+                                             name,
+                                             labels = NULL,
+                                             include.lowest = FALSE,
+                                             right = TRUE,
+                                             dig.lab = 3,
+                                             ordered_result = FALSE, ...){
+    if (missing(name)) {
+        halt("Must provide the name for the new variable")
+    }
+    if (length(breaks) == 1L) {
+        if (is.na(breaks) || breaks < 2L) {
+            halt("invalid number of intervals")
```

Comment:
I know this is verbatim what `base::cut.default` says, but the argument is called "breaks" and the message says "intervals"